I'm getting a build error trying to test:

First, lots of messages like this:

> In file included from ./chainparams.h:10,
>                  from bitcoind.cpp:10:
> ./consensus/params.h: In function ‘constexpr bool Consensus::ValidDeployment(Consensus::BuriedDeployment)’:
> ./consensus/params.h:23:85: warning: comparison is always true due to limited range of data type [-Wtype-limits]
>  constexpr bool ValidDeployment(BuriedDeployment dep) { return DEPLOYMENT_HEIGHTINCB <= dep && dep <= DEPLOYMENT_SEGWIT; }
>                                                                ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
> ./consensus/params.h: In function ‘constexpr bool Consensus::ValidDeployment(Consensus::DeploymentPos)’:
> ./consensus/params.h:32:81: warning: comparison is always true due to limited range of data type [-Wtype-limits]
>  constexpr bool ValidDeployment(DeploymentPos dep) { return DEPLOYMENT_TESTDUMMY <= dep && dep <= DEPLOYMENT_TAPROOT; }
> 
> In file included from ./chainparams.h:10,
>                  from addrdb.cpp:9:
> ./consensus/params.h: In function ‘constexpr bool Consensus::ValidDeployment(Consensus::BuriedDeployment)’:
> ./consensus/params.h:23:85: warning: comparison is always true due to limited range of data type [-Wtype-limits]
>  constexpr bool ValidDeployment(BuriedDeployment dep) { return DEPLOYMENT_HEIGHTINCB <= dep && dep <= DEPLOYMENT_SEGWIT; }
>                                                                ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
> ./consensus/params.h: In function ‘constexpr bool Consensus::ValidDeployment(Consensus::DeploymentPos)’:
> ./consensus/params.h:32:81: warning: comparison is always true due to limited range of data type [-Wtype-limits]
>  constexpr bool ValidDeployment(DeploymentPos dep) { return DEPLOYMENT_TESTDUMMY <= dep && dep <= DEPLOYMENT_TAPROOT; }
> 
> 
> In file included from ./chain.h:10,
>                  from ./deploymentstatus.h:8,
>                  from deploymentstatus.cpp:5:
> ./consensus/params.h: In function ‘constexpr bool Consensus::ValidDeployment(Consensus::BuriedDeployment)’:
> ./consensus/params.h:23:85: warning: comparison is always true due to limited range of data type [-Wtype-limits]
>  constexpr bool ValidDeployment(BuriedDeployment dep) { return DEPLOYMENT_HEIGHTINCB <= dep && dep <= DEPLOYMENT_SEGWIT; }
>                                                                ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
> ./consensus/params.h: In function ‘constexpr bool Consensus::ValidDeployment(Consensus::DeploymentPos)’:
> ./consensus/params.h:32:81: warning: comparison is always true due to limited range of data type [-Wtype-limits]
>  constexpr bool ValidDeployment(DeploymentPos dep) { return DEPLOYMENT_TESTDUMMY <= dep && dep <= DEPLOYMENT_TAPROOT; }


Then this final error message:

> 
> /usr/bin/ld: libbitcoin_server.a(libbitcoin_server_a-ui_interface.o): in function `_GLOBAL__sub_I_00100_0_uiInterface':
> /home/user/bitcoin/src/node/ui_interface.cpp:65: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_server.a(libbitcoin_server_a-ui_interface.o): in function `_GLOBAL__sub_D_00100_1_uiInterface':
> /home/user/bitcoin/src/node/ui_interface.cpp:65: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_server.a(libbitcoin_server_a-ui_interface.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_wallet.a(libbitcoin_wallet_a-context.o): in function `_GLOBAL__sub_I_00100_0__ZN13WalletContextC2Ev':
> /home/user/bitcoin/src/wallet/context.cpp:8: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_wallet.a(libbitcoin_wallet_a-context.o): in function `_GLOBAL__sub_D_00100_1__ZN13WalletContextC2Ev':
> /home/user/bitcoin/src/wallet/context.cpp:8: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_wallet.a(libbitcoin_wallet_a-context.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_common.a(libbitcoin_common_a-bech32.o): in function `_GLOBAL__sub_I_00100_0__ZN6bech326EncodeENS_8EncodingERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorIhSaIhEE':
> /home/user/bitcoin/src/bech32.cpp:198: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_common.a(libbitcoin_common_a-bech32.o): in function `_GLOBAL__sub_D_00100_1__ZN6bech326EncodeENS_8EncodingERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt6vectorIhSaIhEE':
> /home/user/bitcoin/src/bech32.cpp:198: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_common.a(libbitcoin_common_a-bech32.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-glibcxx_sanity.o): in function `_GLOBAL__sub_I_00100_0__Z21sanity_test_range_fmtv':
> /home/user/bitcoin/src/compat/glibcxx_sanity.cpp:61: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-glibcxx_sanity.o): in function `_GLOBAL__sub_D_00100_1__Z21sanity_test_range_fmtv':
> /home/user/bitcoin/src/compat/glibcxx_sanity.cpp:61: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-glibcxx_sanity.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-fs.o): in function `_GLOBAL__sub_I_00100_0__ZN8fsbridge5fopenERKN5boost10filesystem4pathEPKc':
> /home/user/bitcoin/src/fs.cpp:151: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-fs.o): in function `_GLOBAL__sub_D_00100_1__ZN8fsbridge5fopenERKN5boost10filesystem4pathEPKc':
> /home/user/bitcoin/src/fs.cpp:151: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-fs.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-handler.o): in function `_GLOBAL__sub_I_00100_0__ZN10interfaces11MakeHandlerEN5boost8signals210connectionE':
> /home/user/bitcoin/src/interfaces/handler.cpp:14: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-handler.o): in function `_GLOBAL__sub_D_00100_1__ZN10interfaces11MakeHandlerEN5boost8signals210connectionE':
> /home/user/bitcoin/src/interfaces/handler.cpp:14: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-handler.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-cleanse.o): in function `_GLOBAL__sub_I_00100_0__Z14memory_cleansePvm':
> /home/user/bitcoin/src/support/cleanse.cpp:35: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-cleanse.o): in function `_GLOBAL__sub_D_00100_1__Z14memory_cleansePvm':
> /home/user/bitcoin/src/support/cleanse.cpp:35: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-cleanse.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-bip32.o): in function `_GLOBAL__sub_I_00100_0__Z14ParseHDKeypathRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt6vectorIjSaIjEE':
> /home/user/bitcoin/src/util/bip32.cpp:66: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-bip32.o): in function `_GLOBAL__sub_D_00100_1__Z14ParseHDKeypathRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSt6vectorIjSaIjEE':
> /home/user/bitcoin/src/util/bip32.cpp:66: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-bip32.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-moneystr.o): in function `_GLOBAL__sub_I_00100_0__Z11FormatMoneyB5cxx11l':
> /home/user/bitcoin/src/util/moneystr.cpp:82: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-moneystr.o): in function `_GLOBAL__sub_D_00100_1__Z11FormatMoneyB5cxx11l':
> /home/user/bitcoin/src/util/moneystr.cpp:82: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-moneystr.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-readwritefile.o): in function `_GLOBAL__sub_I_00100_0__Z14ReadBinaryFileB5cxx11RKN5boost10filesystem4pathEm':
> /home/user/bitcoin/src/util/readwritefile.cpp:32: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-readwritefile.o): in function `_GLOBAL__sub_D_00100_1__Z14ReadBinaryFileB5cxx11RKN5boost10filesystem4pathEm':
> /home/user/bitcoin/src/util/readwritefile.cpp:32: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-readwritefile.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-settings.o): in function `_GLOBAL__sub_I_00100_0__ZN4util12ReadSettingsERKN5boost10filesystem4pathERSt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE8UniValueSt4lessISB_ESaISt4pairIKSB_SC_EEERSt6vectorISB_SaISB_EE':
> /home/user/bitcoin/src/util/settings.cpp:240: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-settings.o): in function `_GLOBAL__sub_D_00100_1__ZN4util12ReadSettingsERKN5boost10filesystem4pathERSt3mapINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE8UniValueSt4lessISB_ESaISt4pairIKSB_SC_EEERSt6vectorISB_SaISB_EE':
> /home/user/bitcoin/src/util/settings.cpp:240: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-settings.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-url.o): in function `_GLOBAL__sub_I_00100_0__Z9urlDecodeRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE':
> /home/user/bitcoin/src/util/url.cpp:21: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-url.o): in function `_GLOBAL__sub_D_00100_1__Z9urlDecodeRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE':
> /home/user/bitcoin/src/util/url.cpp:21: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_util.a(libbitcoin_util_a-url.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: libbitcoin_consensus.a(libbitcoin_consensus_a-script_error.o): in function `_GLOBAL__sub_I_00100_0__Z17ScriptErrorStringB5cxx1113ScriptError_t':
> /home/user/bitcoin/src/script/script_error.cpp:123: undefined reference to `__gcov_init'
> /usr/bin/ld: libbitcoin_consensus.a(libbitcoin_consensus_a-script_error.o): in function `_GLOBAL__sub_D_00100_1__Z17ScriptErrorStringB5cxx1113ScriptError_t':
> /home/user/bitcoin/src/script/script_error.cpp:123: undefined reference to `__gcov_exit'
> /usr/bin/ld: libbitcoin_consensus.a(libbitcoin_consensus_a-script_error.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-aes.o): in function `_GLOBAL__sub_I_00100_0_AES128_init':
> /home/user/bitcoin/src/crypto/aes.cpp:152: undefined reference to `__gcov_init'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-aes.o): in function `_GLOBAL__sub_D_00100_1_AES128_init':
> /home/user/bitcoin/src/crypto/aes.cpp:152: undefined reference to `__gcov_exit'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-aes.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-hmac_sha256.o): in function `_GLOBAL__sub_I_00100_0__ZN12CHMAC_SHA256C2EPKhm':
> /home/user/bitcoin/src/crypto/hmac_sha256.cpp:34: undefined reference to `__gcov_init'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-hmac_sha256.o): in function `_GLOBAL__sub_D_00100_1__ZN12CHMAC_SHA256C2EPKhm':
> /home/user/bitcoin/src/crypto/hmac_sha256.cpp:34: undefined reference to `__gcov_exit'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-hmac_sha256.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-hmac_sha512.o): in function `_GLOBAL__sub_I_00100_0__ZN12CHMAC_SHA512C2EPKhm':
> /home/user/bitcoin/src/crypto/hmac_sha512.cpp:34: undefined reference to `__gcov_init'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-hmac_sha512.o): in function `_GLOBAL__sub_D_00100_1__ZN12CHMAC_SHA512C2EPKhm':
> /home/user/bitcoin/src/crypto/hmac_sha512.cpp:34: undefined reference to `__gcov_exit'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-hmac_sha512.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-sha256_sse4.o): in function `_GLOBAL__sub_I_00100_0__ZN11sha256_sse49TransformEPjPKhm':
> /home/user/bitcoin/src/crypto/sha256_sse4.cpp:956: undefined reference to `__gcov_init'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-sha256_sse4.o): in function `_GLOBAL__sub_D_00100_1__ZN11sha256_sse49TransformEPjPKhm':
> /home/user/bitcoin/src/crypto/sha256_sse4.cpp:956: undefined reference to `__gcov_exit'
> /usr/bin/ld: crypto/libbitcoin_crypto_base.a(libbitcoin_crypto_base_a-sha256_sse4.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: crypto/libbitcoin_crypto_shani.a(libbitcoin_crypto_shani_a-sha256_shani.o): in function `_GLOBAL__sub_I_00100_0__ZN12sha256_shani9TransformEPjPKhm':
> /home/user/bitcoin/src/crypto/sha256_shani.cpp:352: undefined reference to `__gcov_init'
> /usr/bin/ld: crypto/libbitcoin_crypto_shani.a(libbitcoin_crypto_shani_a-sha256_shani.o): in function `_GLOBAL__sub_D_00100_1__ZN12sha256_shani9TransformEPjPKhm':
> /home/user/bitcoin/src/crypto/sha256_shani.cpp:352: undefined reference to `__gcov_exit'
> /usr/bin/ld: crypto/libbitcoin_crypto_shani.a(libbitcoin_crypto_shani_a-sha256_shani.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-db_impl.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb15SanitizeOptionsERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPKNS_21InternalKeyComparatorEPKNS_20InternalFilterPolicyERKNS_7OptionsE':
> /home/user/bitcoin/src/leveldb/db/db_impl.cc:1104: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-db_impl.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb15SanitizeOptionsERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPKNS_21InternalKeyComparatorEPKNS_20InternalFilterPolicyERKNS_7OptionsE':
> /home/user/bitcoin/src/leveldb/db/db_impl.cc:1104: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-db_impl.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-db_iter.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb13NewDBIteratorEPNS_6DBImplEPKNS_10ComparatorEPNS_8IteratorEmj':
> /home/user/bitcoin/src/leveldb/db/db_iter.cc:175: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-db_iter.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb13NewDBIteratorEPNS_6DBImplEPKNS_10ComparatorEPNS_8IteratorEmj':
> /home/user/bitcoin/src/leveldb/db/db_iter.cc:175: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-db_iter.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-filename.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb11LogFileNameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEm':
> /home/user/bitcoin/src/leveldb/db/filename.cc:139: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-filename.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb11LogFileNameERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEm':
> /home/user/bitcoin/src/leveldb/db/filename.cc:139: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-filename.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-log_reader.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb3log6Reader8ReporterD2Ev':
> /home/user/bitcoin/src/leveldb/db/log_reader.cc:174: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-log_reader.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb3log6Reader8ReporterD2Ev':
> /home/user/bitcoin/src/leveldb/db/log_reader.cc:174: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-log_reader.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-log_writer.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb3log6WriterC2EPNS_12WritableFileE':
> /home/user/bitcoin/src/leveldb/db/log_writer.cc:80: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-log_writer.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb3log6WriterC2EPNS_12WritableFileE':
> /home/user/bitcoin/src/leveldb/db/log_writer.cc:80: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-log_writer.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-memtable.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb8MemTableC2ERKNS_21InternalKeyComparatorE':
> /home/user/bitcoin/src/leveldb/db/memtable.cc:135: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-memtable.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb8MemTableC2ERKNS_21InternalKeyComparatorE':
> /home/user/bitcoin/src/leveldb/db/memtable.cc:135: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-memtable.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-table_cache.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb10TableCacheC2ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKNS_7OptionsEi':
> /home/user/bitcoin/src/leveldb/db/table_cache.cc:37: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-table_cache.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb10TableCacheC2ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKNS_7OptionsEi':
> /home/user/bitcoin/src/leveldb/db/table_cache.cc:37: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-table_cache.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-version_edit.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb11VersionEdit5ClearEv':
> /home/user/bitcoin/src/leveldb/db/version_edit.cc:203: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-version_edit.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb11VersionEdit5ClearEv':
> /home/user/bitcoin/src/leveldb/db/version_edit.cc:203: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-version_edit.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-version_set.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb7VersionD2Ev':
> /home/user/bitcoin/src/leveldb/db/version_set.cc:246: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-version_set.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb7VersionD2Ev':
> /home/user/bitcoin/src/leveldb/db/version_set.cc:246: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-version_set.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-write_batch.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb10WriteBatchC2Ev':
> /home/user/bitcoin/src/leveldb/db/write_batch.cc:29: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-write_batch.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb10WriteBatchC2Ev':
> /home/user/bitcoin/src/leveldb/db/write_batch.cc:29: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-write_batch.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-iterator.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb8IteratorC2Ev':
> /home/user/bitcoin/src/leveldb/table/iterator.cc:74: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-iterator.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb8IteratorC2Ev':
> /home/user/bitcoin/src/leveldb/table/iterator.cc:74: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-iterator.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-merger.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb18NewMergingIteratorEPKNS_10ComparatorEPPNS_8IteratorEi':
> /home/user/bitcoin/src/leveldb/table/merger.cc:189: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-merger.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb18NewMergingIteratorEPKNS_10ComparatorEPPNS_8IteratorEi':
> /home/user/bitcoin/src/leveldb/table/merger.cc:189: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-merger.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-table_builder.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb12TableBuilderC2ERKNS_7OptionsEPNS_12WritableFileE':
> /home/user/bitcoin/src/leveldb/table/table_builder.cc:123: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-table_builder.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb12TableBuilderC2ERKNS_7OptionsEPNS_12WritableFileE':
> /home/user/bitcoin/src/leveldb/table/table_builder.cc:123: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-table_builder.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-table.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb5Table4OpenERKNS_7OptionsEPNS_16RandomAccessFileEmPPS0_':
> /home/user/bitcoin/src/leveldb/table/table.cc:82: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-table.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb5Table4OpenERKNS_7OptionsEPNS_16RandomAccessFileEmPPS0_':
> /home/user/bitcoin/src/leveldb/table/table.cc:82: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-table.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-two_level_iterator.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb19NewTwoLevelIteratorEPNS_8IteratorEPFS1_PvRKNS_11ReadOptionsERKNS_5SliceEES2_S5_':
> /home/user/bitcoin/src/leveldb/table/two_level_iterator.cc:87: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-two_level_iterator.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb19NewTwoLevelIteratorEPNS_8IteratorEPFS1_PvRKNS_11ReadOptionsERKNS_5SliceEES2_S5_':
> /home/user/bitcoin/src/leveldb/table/two_level_iterator.cc:87: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-two_level_iterator.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-arena.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb5ArenaC2Ev':
> /home/user/bitcoin/src/leveldb/util/arena.cc:56: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-arena.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb5ArenaC2Ev':
> /home/user/bitcoin/src/leveldb/util/arena.cc:56: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-arena.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-bloom.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb20NewBloomFilterPolicyEi':
> /home/user/bitcoin/src/leveldb/util/bloom.cc:54: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-bloom.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb20NewBloomFilterPolicyEi':
> /home/user/bitcoin/src/leveldb/util/bloom.cc:54: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-bloom.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-cache.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb5CacheD2Ev':
> /home/user/bitcoin/src/leveldb/util/cache.cc:398: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-cache.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb5CacheD2Ev':
> /home/user/bitcoin/src/leveldb/util/cache.cc:398: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-cache.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-coding.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb10PutFixed32EPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEj':
> /home/user/bitcoin/src/leveldb/util/coding.cc:70: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-coding.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb10PutFixed32EPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEj':
> /home/user/bitcoin/src/leveldb/util/coding.cc:70: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-coding.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-comparator.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb10ComparatorD2Ev':
> /home/user/bitcoin/src/leveldb/util/comparator.cc:52: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-comparator.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb10ComparatorD2Ev':
> /home/user/bitcoin/src/leveldb/util/comparator.cc:52: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-comparator.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-crc32c.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb6crc32c6ExtendEjPKcm':
> /home/user/bitcoin/src/leveldb/util/crc32c.cc:377: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-crc32c.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb6crc32c6ExtendEjPKcm':
> /home/user/bitcoin/src/leveldb/util/crc32c.cc:377: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-crc32c.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-env.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb3EnvD2Ev':
> /home/user/bitcoin/src/leveldb/util/env.cc:88: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-env.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb3EnvD2Ev':
> /home/user/bitcoin/src/leveldb/util/env.cc:88: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-env.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-filter_policy.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb12FilterPolicyD2Ev':
> /home/user/bitcoin/src/leveldb/util/filter_policy.cc:9: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-filter_policy.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb12FilterPolicyD2Ev':
> /home/user/bitcoin/src/leveldb/util/filter_policy.cc:9: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-filter_policy.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-hash.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb4HashEPKcmj':
> /home/user/bitcoin/src/leveldb/util/hash.cc:53: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-hash.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb4HashEPKcmj':
> /home/user/bitcoin/src/leveldb/util/hash.cc:53: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-hash.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-logging.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb14AppendNumberToEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEm':
> /home/user/bitcoin/src/leveldb/util/logging.cc:49: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-logging.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb14AppendNumberToEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEm':
> /home/user/bitcoin/src/leveldb/util/logging.cc:49: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-logging.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-options.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb7OptionsC2Ev':
> /home/user/bitcoin/src/leveldb/util/options.cc:12: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-options.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb7OptionsC2Ev':
> /home/user/bitcoin/src/leveldb/util/options.cc:12: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-options.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-status.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb6Status9CopyStateEPKc':
> /home/user/bitcoin/src/leveldb/util/status.cc:75: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-status.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb6Status9CopyStateEPKc':
> /home/user/bitcoin/src/leveldb/util/status.cc:75: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-status.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-env_posix.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb18EnvPosixTestHelper18SetReadOnlyFDLimitEi':
> /home/user/bitcoin/src/leveldb/util/env_posix.cc:674: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-env_posix.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb18EnvPosixTestHelper18SetReadOnlyFDLimitEi':
> /home/user/bitcoin/src/leveldb/util/env_posix.cc:674: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-env_posix.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-builder.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb10BuildTableERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPNS_3EnvERKNS_7OptionsEPNS_10TableCacheEPNS_8IteratorEPNS_12FileMetaDataE':
> /home/user/bitcoin/src/leveldb/db/builder.cc:77: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-builder.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb10BuildTableERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPNS_3EnvERKNS_7OptionsEPNS_10TableCacheEPNS_8IteratorEPNS_12FileMetaDataE':
> /home/user/bitcoin/src/leveldb/db/builder.cc:77: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-builder.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-dbformat.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb17AppendInternalKeyEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKNS_17ParsedInternalKeyE':
> /home/user/bitcoin/src/leveldb/db/dbformat.cc:42: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-dbformat.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb17AppendInternalKeyEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKNS_17ParsedInternalKeyE':
> /home/user/bitcoin/src/leveldb/db/dbformat.cc:42: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-dbformat.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-block_builder.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb12BlockBuilderC2EPKNS_7OptionsE':
> /home/user/bitcoin/src/leveldb/table/block_builder.cc:45: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-block_builder.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb12BlockBuilderC2EPKNS_7OptionsE':
> /home/user/bitcoin/src/leveldb/table/block_builder.cc:45: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-block_builder.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-block.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb5BlockC2ERKNS_13BlockContentsE':
> /home/user/bitcoin/src/leveldb/table/block.cc:265: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-block.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb5BlockC2ERKNS_13BlockContentsE':
> /home/user/bitcoin/src/leveldb/table/block.cc:265: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-block.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-filter_block.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb18FilterBlockBuilderC2EPKNS_12FilterPolicyE':
> /home/user/bitcoin/src/leveldb/table/filter_block.cc:27: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-filter_block.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb18FilterBlockBuilderC2EPKNS_12FilterPolicyE':
> /home/user/bitcoin/src/leveldb/table/filter_block.cc:27: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-filter_block.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-format.o): in function `_GLOBAL__sub_I_00100_0__ZNK7leveldb11BlockHandle8EncodeToEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE':
> /home/user/bitcoin/src/leveldb/table/format.cc:40: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-format.o): in function `_GLOBAL__sub_D_00100_1__ZNK7leveldb11BlockHandle8EncodeToEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE':
> /home/user/bitcoin/src/leveldb/table/format.cc:40: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libleveldb.a(libleveldb_a-format.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: crc32c/libcrc32c.a(libcrc32c_a-crc32c.o): in function `_GLOBAL__sub_I_00100_0__ZN6crc32c6ExtendEjPKhm':
> /home/user/bitcoin/src/crc32c/src/crc32c.cc:37: undefined reference to `__gcov_init'
> /usr/bin/ld: crc32c/libcrc32c.a(libcrc32c_a-crc32c.o): in function `_GLOBAL__sub_D_00100_1__ZN6crc32c6ExtendEjPKhm':
> /home/user/bitcoin/src/crc32c/src/crc32c.cc:37: undefined reference to `__gcov_exit'
> /usr/bin/ld: crc32c/libcrc32c.a(libcrc32c_a-crc32c.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: crc32c/libcrc32c.a(libcrc32c_a-crc32c_portable.o): in function `_GLOBAL__sub_I_00100_0__ZN6crc32c14ExtendPortableEjPKhm':
> /home/user/bitcoin/src/crc32c/src/crc32c_portable.cc:349: undefined reference to `__gcov_init'
> /usr/bin/ld: crc32c/libcrc32c.a(libcrc32c_a-crc32c_portable.o): in function `_GLOBAL__sub_D_00100_1__ZN6crc32c14ExtendPortableEjPKhm':
> /home/user/bitcoin/src/crc32c/src/crc32c_portable.cc:349: undefined reference to `__gcov_exit'
> /usr/bin/ld: crc32c/libcrc32c.a(libcrc32c_a-crc32c_portable.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: crc32c/libcrc32c_sse42.a(libcrc32c_sse42_a-crc32c_sse42.o): in function `_GLOBAL__sub_I_00100_0__ZN6crc32c11ExtendSse42EjPKhm':
> /home/user/bitcoin/src/crc32c/src/crc32c_sse42.cc:254: undefined reference to `__gcov_init'
> /usr/bin/ld: crc32c/libcrc32c_sse42.a(libcrc32c_sse42_a-crc32c_sse42.o): in function `_GLOBAL__sub_D_00100_1__ZN6crc32c11ExtendSse42EjPKhm':
> /home/user/bitcoin/src/crc32c/src/crc32c_sse42.cc:254: undefined reference to `__gcov_exit'
> /usr/bin/ld: crc32c/libcrc32c_sse42.a(libcrc32c_sse42_a-crc32c_sse42.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> /usr/bin/ld: leveldb/libmemenv.a(libmemenv_a-memenv.o): in function `_GLOBAL__sub_I_00100_0__ZN7leveldb9NewMemEnvEPNS_3EnvE':
> /home/user/bitcoin/src/leveldb/helpers/memenv/memenv.cc:209: undefined reference to `__gcov_init'
> /usr/bin/ld: leveldb/libmemenv.a(libmemenv_a-memenv.o): in function `_GLOBAL__sub_D_00100_1__ZN7leveldb9NewMemEnvEPNS_3EnvE':
> /home/user/bitcoin/src/leveldb/helpers/memenv/memenv.cc:209: undefined reference to `__gcov_exit'
> /usr/bin/ld: leveldb/libmemenv.a(libmemenv_a-memenv.o):(.data.rel+0x20): undefined reference to `__gcov_merge_add'
> collect2: error: ld returned 1 exit status
> make[2]: *** [Makefile:5923: bitcoind] Error 1
> make[2]: Leaving directory '/home/user/bitcoin/src'
> make[1]: *** [Makefile:16161: all-recursive] Error 1
> make[1]: Leaving directory '/home/user/bitcoin/src'
> make: *** [Makefile:820: all-recursive] Error 1

--- a/src/rpc/client.cpp
+++ b/src/rpc/client.cpp
@@ -142,6 +142,7 @@ static const CRPCConvertParam vRPCConvertParams[] =
     { "importmulti", 0, "requests" },
     { "importmulti", 1, "options" },
     { "importdescriptors", 0, "requests" },
+    { "listdescriptors", 0, "private" },
     { "verifychain", 0, "checklevel" },
     { "verifychain", 1, "nblocks" },
     { "getblockstats", 0, "hash_or_height" },

--- a/src/wallet/rpcdump.cpp
+++ b/src/wallet/rpcdump.cpp
@@ -1755,8 +1755,10 @@ RPCHelpMan listdescriptors()
 {
     return RPCHelpMan{
         "listdescriptors",
-        "\nList descriptors imported into a descriptor-enabled wallet.",
-        {},
+        "\nList descriptors imported into a descriptor-enabled wallet.\n",
+        {
+            {"private", RPCArg::Type::BOOL, RPCArg::Default{false}, "Show private descriptors."}
+        },
         RPCResult{RPCResult::Type::OBJ, "", "", {
             {RPCResult::Type::STR, "wallet_name", "Name of wallet this operation was performed on"},
             {RPCResult::Type::ARR, "descriptors", "Array of descriptor objects",
@@ -1776,6 +1778,7 @@ RPCHelpMan listdescriptors()
         }},
         RPCExamples{
             HelpExampleCli("listdescriptors", "") + HelpExampleRpc("listdescriptors", "")
+            + HelpExampleCli("listdescriptors", "true") + HelpExampleRpc("listdescriptors", "true")
         },
         [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
 {
@@ -1784,6 +1787,11 @@ RPCHelpMan listdescriptors()
 
     if (!wallet->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS)) {
         throw JSONRPCError(RPC_WALLET_ERROR, "listdescriptors is not available for non-descriptor wallets");
+    }
+
+    const bool priv = !request.params[0].isNull() && request.params[0].get_bool();
+    if (priv) {
+        EnsureWalletIsUnlocked(*wallet);
     }
 
     LOCK(wallet->cs_wallet);
@@ -1799,8 +1807,9 @@ RPCHelpMan listdescriptors()
         LOCK(desc_spk_man->cs_desc_man);
         const auto& wallet_descriptor = desc_spk_man->GetWalletDescriptor();
         std::string descriptor;
-        if (!desc_spk_man->GetDescriptorString(descriptor)) {
-            throw JSONRPCError(RPC_WALLET_ERROR, "Can't get normalized descriptor string.");
+
+        if (!desc_spk_man->GetDescriptorString(descriptor, priv)) {
+            throw JSONRPCError(RPC_WALLET_ERROR, "Can't get descriptor string.");
         }
         spk.pushKV("desc", descriptor);
         spk.pushKV("timestamp", wallet_descriptor.creation_time);

--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -3872,7 +3872,7 @@ RPCHelpMan getaddressinfo()
     DescriptorScriptPubKeyMan* desc_spk_man = dynamic_cast<DescriptorScriptPubKeyMan*>(pwallet->GetScriptPubKeyMan(scriptPubKey));
     if (desc_spk_man) {
         std::string desc_str;
-        if (desc_spk_man->GetDescriptorString(desc_str)) {
+        if (desc_spk_man->GetDescriptorString(desc_str, /* priv */ false)) {
             ret.pushKV("parent_desc", desc_str);
         }
     }

--- a/src/wallet/scriptpubkeyman.cpp
+++ b/src/wallet/scriptpubkeyman.cpp
@@ -2258,12 +2258,19 @@ const std::vector<CScript> DescriptorScriptPubKeyMan::GetScriptPubKeys() const
     return script_pub_keys;
 }
 
-bool DescriptorScriptPubKeyMan::GetDescriptorString(std::string& out) const
+bool DescriptorScriptPubKeyMan::GetDescriptorString(std::string& out, const bool priv) const
 {
     LOCK(cs_desc_man);
 
     FlatSigningProvider provider;
     provider.keys = GetKeys();
+
+    if (priv) {
+        // For the private version, always return the master key to avoid
+        // exposing child private keys. The risk implications of exposing child
+        // private keys together with the parent xpub may be non-obvious for users.
+        return m_wallet_descriptor.descriptor->ToPrivateString(provider, out);
+    }
 
     return m_wallet_descriptor.descriptor->ToNormalizedString(provider, out, &m_wallet_descriptor.cache);
 }

--- a/src/wallet/scriptpubkeyman.h
+++ b/src/wallet/scriptpubkeyman.h
@@ -621,7 +621,7 @@ public:
     const WalletDescriptor GetWalletDescriptor() const EXCLUSIVE_LOCKS_REQUIRED(cs_desc_man);
     const std::vector<CScript> GetScriptPubKeys() const;
 
-    bool GetDescriptorString(std::string& out) const;
+    bool GetDescriptorString(std::string& out, const bool priv) const;
 
     void UpgradeDescriptorCache();
 };

--- a/test/functional/wallet_listdescriptors.py
+++ b/test/functional/wallet_listdescriptors.py
@@ -71,10 +71,38 @@ class ListDescriptorsTest(BitcoinTestFramework):
             ],
         }
         assert_equal(expected, wallet.listdescriptors())
+        assert_equal(expected, wallet.listdescriptors(False))
+
+        self.log.info('Test list private descriptors')
+        expected_private = {
+            'wallet_name': 'w2',
+            'descriptors': [
+                {'desc': descsum_create('wpkh(' + xprv + hardened_path + '/0/*)'),
+                 'timestamp': 1296688602,
+                 'active': False,
+                 'range': [0, 0],
+                 'next': 0},
+            ],
+        }
+        assert_equal(expected_private, wallet.listdescriptors(True))
 
         self.log.info("Test listdescriptors with encrypted wallet")
         wallet.encryptwallet("pass")
         assert_equal(expected, wallet.listdescriptors())
+
+        self.log.info('Test list private descriptors with encrypted wallet')
+        assert_raises_rpc_error(-13, 'Please enter the wallet passphrase with walletpassphrase first.', wallet.listdescriptors, True)
+        wallet.walletpassphrase(passphrase="pass", timeout=1000000)
+        assert_equal(expected_private, wallet.listdescriptors(True))
+
+        self.log.info('Test list private descriptors with watch-only wallet')
+        node.createwallet(wallet_name='watch-only', descriptors=True, disable_private_keys=True)
+        watch_only_wallet = node.get_wallet_rpc('watch-only')
+        watch_only_wallet.importdescriptors([{
+            'desc': descsum_create('wpkh(' + xpub_acc + ')'),
+            'timestamp': 1296688602,
+        }])
+        assert_raises_rpc_error(-4, 'Can\'t get descriptor string', watch_only_wallet.listdescriptors, True)
 
         self.log.info('Test non-active non-range combo descriptor')
         node.createwallet(wallet_name='w4', blank=True, descriptors=True)

